stdlib/src/utils/string_slice.mojo

```diff
@@ -64,6 +65,15 @@ fn _shift_unicode_to_utf8(ptr: UnsafePointer[UInt8], c: Int, num_bytes: Int):
         ptr[i] = ((c >> shift) & 0b0011_1111) | 0b1000_0000
 
 
+@always_inline("nodebug")
+fn _utf8_first_byte_sequence_length(b: UInt8) -> Int:
+    """Get the length of the sequence starting with given byte. Do note that
```

Comment:
**Question** Should we add a `debug_assert` that the given byte is not a continuation byte then?

Reply:
I see the conversation here resolved, but didn't see a `debug_assert` added.  Is that intentional?

Reply:
Yeah I'm just waiting for @msaelices response to add everything in 1 commit